### PR TITLE
added spacy embedding similarity #9

### LINKED ISSUE #9: Allow to provide embeddings (words_and_vectors)
DataAugmenterNLP (augment_column method)  doesnt allow to provide words_and_vectors for "embedding" augmentation type

### PR DESCRIPTION
**Pull request by AlexKay28**
Added spacy engine who handling similarity comparing.
now we only should ask users to load spacy and embedding by himself

### CODE REVIEW

**Comment by AlexKay28** [comment]
@joein , dont take into account that default embedding... its weird actually
but I guess I would ask your opinion about the method "augment_columns" in general. It looks like a part which should be removed. too overloaded i think

File: setup.py [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
-    'pandas',       # pandas==1.1.5
-    "scikit-learn", # scikit-learn==0.24.1
-    "nlpaug",       # nlpaug==1.1.3
-    "wget",         # wget==3.2
+    "numpy",  # numpy==1.18.5
+    "pandas",  # pandas==1.1.5
+    "scikit-learn",  # scikit-learn==0.24.1
+    "nlpaug",  # nlpaug==1.1.3
+    "wget",  # wget==3.2
+    "spacy",  # spacy==3.0.5
```
Comment (joein) [inline setup.py]: we should pin versions

File: zarnitsa/DataAugmenterNLP.py [hunk 2]
```diff
@@ -110,70 +112,79 @@ def _check_synset(self, name):
...
+    def augment_ppdb(self, text: str) -> str:
         """Augment column data using ppdb synset."""
         if not self.__aug_ppdb:
             print("Load PPDB synset")
             self._check_synset("ppdb")
             self.__aug_ppdb = naw.SynonymAug(
                 aug_src="ppdb",
-                model_path=self.__class_local_path
-                + "/internal_data/ppdb-2.0-tldr",
+                model_path=self.__class_local_path + "/internal_data/ppdb-2.0-tldr",
```
Comment (joein) [inline zarnitsa/DataAugmenterNLP.py]: I think we should consider usage of some libraries like Pathlib to operate with such things

File: zarnitsa/DataAugmenterNLP.py [hunk 3]
```diff
@@ -110,70 +112,79 @@ def _check_synset(self, name):
...
+            )
+            return [
+                w.lower_.capitalize()
+                if not word.lower_.islower() and not word.lower_.isupper()
+                else w.lower_
+                if word.islower()
+                else word.lower_.upper()
+                for w in by_similarity[: topn + 1]
+                if w.lower_ != word.lower_
+            ]
```
Comment (joein) [inline zarnitsa/DataAugmenterNLP.py]: I don't think that nested ternary operator is a good style, expand it to achieve better readability

**Review by joein — CHANGES_REQUESTED** [review]
Sorry, I haven't launched it locally yet, I'm going to do it tomorrow, but I left some comments as a first iteration of review

File: setup.py (hunk 1 above)
Comment (AlexKay28) [reply setup.py]: ok, ill pin it

File: zarnitsa/DataAugmenterNLP.py (hunk 2 above)
Comment (AlexKay28) [reply zarnitsa/DataAugmenterNLP.py]: get it

File: zarnitsa/DataAugmenterNLP.py (hunk 3 above)
Comment (AlexKay28) [reply zarnitsa/DataAugmenterNLP.py]: ok

**Review by joein — APPROVED** [review]

**Pull request merged**